geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypes.scala

```diff
@@ -76,7 +76,11 @@ object SimpleFeatureTypes {
     } else {
       Map.empty[String, AnyRef]
     }
-    val opts = userData.map { case (k, v) => new GenericOption(k, v) }.toSeq
+    val opts = userData.map {
+      // Special case to handle adding keywords
+      case (KEYWORDS_KEY, v) => new GenericOption(KEYWORDS_KEY, v.asInstanceOf[java.util.ArrayList[String]].mkString(KEYWORDS_DELIMITER))
+      case (k, v) => new GenericOption(k, v)
+    }.toSeq
```

Comment:
if we're going to do this, then we need to also do the reverse when writing it back out... still not sold on having special handling for the user data though...
question: can we delimit things using the spec string? e.g. normally you can encode user data like https://github.com/locationtech/geomesa/blob/master/geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/index/Z3IdxStrategyTest.scala#L40